snuba/clickhouse/astquery.py

```diff
@@ -44,11 +44,13 @@ def __init__(self, query: Query, settings: RequestSettings,) -> None:
         self.__prewhere = query.get_prewhere_ast()
 
         self.__settings = settings
+        self.__sql_data_list: Optional[List[Tuple[str, str]]] = None
         self.__formatted_query: Optional[str] = None
+        self.__sql_data: Optional[Dict[str, str]] = None
 
-    def _format_query_impl(self) -> str:
-        if self.__formatted_query:
-            return self.__formatted_query
+    def _sql_data_list(self) -> List[Tuple[str, str]]:
+        if self.__sql_data:
+            return self.__sql_data
```

Comment:
returning self.__sql_data would return a dictionary while this method is declared to return a list of tuples.

Reply:
Damn. Thank you.